src/module_3/assign_3.md

```diff
@@ -0,0 +1,980 @@
+      <td>0</td>
+      <td>33826472919172</td>
+      <td>2807985930372</td>
+      <td>3482464092292</td>
+      <td>128098</td>
+      <td>1.601856e+09</td>
+      <td>3</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>...</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>2.0</td>
+      <td>33.0</td>
+      <td>42.0</td>
+      <td>31.134053</td>
+      <td>30.0</td>
+      <td>30.0</td>
+      <td>24.27618</td>
+    </tr>
+    <tr>
+      <th>1</th>
+      <td>0</td>
+      <td>33826472919172</td>
+      <td>2808027644036</td>
+      <td>3466586718340</td>
+      <td>128098</td>
+      <td>1.601856e+09</td>
+      <td>2</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>...</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>2.0</td>
+      <td>33.0</td>
+      <td>42.0</td>
+      <td>31.134053</td>
+      <td>30.0</td>
+      <td>30.0</td>
+      <td>24.27618</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>0</td>
+      <td>33826472919172</td>
+      <td>2808099078276</td>
+      <td>3481384026244</td>
+      <td>128098</td>
+      <td>1.601856e+09</td>
+      <td>4</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>...</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>2.0</td>
+      <td>33.0</td>
+      <td>42.0</td>
+      <td>31.134053</td>
+      <td>30.0</td>
+      <td>30.0</td>
+      <td>24.27618</td>
+    </tr>
+    <tr>
+      <th>3</th>
+      <td>0</td>
+      <td>33826472919172</td>
+      <td>2808393957508</td>
+      <td>3291363377284</td>
+      <td>128098</td>
+      <td>1.601942e+09</td>
+      <td>2</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>...</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>2.0</td>
+      <td>33.0</td>
+      <td>42.0</td>
+      <td>31.134053</td>
+      <td>30.0</td>
+      <td>30.0</td>
+      <td>24.27618</td>
+    </tr>
+    <tr>
+      <th>4</th>
+      <td>0</td>
+      <td>33826472919172</td>
+      <td>2808429314180</td>
+      <td>3537167515780</td>
+      <td>128098</td>
+      <td>1.601942e+09</td>
+      <td>3</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>...</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>0.0</td>
+      <td>2.0</td>
+      <td>33.0</td>
+      <td>42.0</td>
+      <td>31.134053</td>
+      <td>30.0</td>
+      <td>30.0</td>
+      <td>24.27618</td>
+    </tr>
+  </tbody>
+</table>
+<p>5 rows × 26 columns</p>
+</div>
+
+
+
+
+```python
+#Filtering for transactions with more than 5 items sold
+df_orders = df.groupby(by='order_id')['outcome'].sum()
+print(len(df_orders))
+transac_5_items = df_orders[df_orders>5].index.to_list()
+```
+
+    3446
+
+
+
+```python
+#Splitting for test/train/validation, keeping the transactions in the same dataset to avoid data leakeage
+
+random.shuffle(transac_5_items)
```

Comment:
As we saw in class, this can be problematic since it doesn't account for the potential temporal trend in the data.

Reply:
Yeah, changed as well!